internal/services/recoveryservices/recovery_services_vault_data_source.go

```diff
@@ -4,78 +4,113 @@
 package recoveryservices
 
 import (
+	"context"
 	"fmt"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-helpers/lang/response"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
-	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/recoveryservices/2024-01-01/vaults"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
 )
 
-func dataSourceRecoveryServicesVault() *pluginsdk.Resource {
-	return &pluginsdk.Resource{
-		Read: dataSourceRecoveryServicesVaultRead,
+var _ sdk.DataSource = SiteRecoveryRecoveryVaultDataSource{}
 
-		Timeouts: &pluginsdk.ResourceTimeout{
-			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
-		},
+type SiteRecoveryRecoveryVaultDataSource struct{}
+type SiteRecoveryRecoveryVaultDataSourceModel struct {
+	Name              string                                     `tfschema:"name"`
+	RecoveryGroupName string                                     `tfschema:"resource_group_name"`
```

Comment:
Minor but attribute names in the struct should reflect the property name in the schema in most cases. I'm assuming this is a typo/accident?
```suggestion
	ResourceGroupName string                                     `tfschema:"resource_group_name"`
```

Reply:
Yes, sorry for this.